packages/headless/src/features/case-assist/case-assist-analytics-selectors.ts

```diff
@@ -0,0 +1,124 @@
+import {CaseAssistAppState} from '../../state/case-assist-app-state';
+
+export enum KnownCaseFields {
+  id = 'id',
+  subject = 'subject',
+  description = 'description',
+  category = 'category',
+  productId = 'productId',
+}
+
+const isCustomFieldName = (fieldName: string) =>
+  !(Object.values(KnownCaseFields) as string[]).includes(fieldName);
+
+export const selectCase = (state: Partial<CaseAssistAppState>) => {
+  const selectFieldValue = (fieldName: string) =>
+    selectCaseInputValue(state, fieldName) ??
+    selectCaseFieldValue(state, fieldName);
+
+  return {
+    id: selectFieldValue(KnownCaseFields.id),
+    subject: selectFieldValue(KnownCaseFields.subject),
+    description: selectFieldValue(KnownCaseFields.description),
+    category: selectFieldValue(KnownCaseFields.category),
+    productId: selectFieldValue(KnownCaseFields.productId),
+    custom: {
+      ...selectCustomCaseInputValues(state),
+      ...selectCustomCaseFieldValues(state),
+    },
+  };
+};
+
+export const selectCaseInputValue = (
+  state: Partial<CaseAssistAppState>,
+  fieldName: string
+) => state?.caseInput?.[fieldName]?.value;
+
+export const selectCustomCaseInputValues = (
+  state: Partial<CaseAssistAppState>
+) =>
+  Object.keys(state?.caseInput ?? []).reduce((customFields, fieldName) => {
+    if (isCustomFieldName(fieldName)) {
+      const value = state?.caseInput?.[fieldName]?.value;
+      if (typeof value !== 'undefined') {
+        return {...customFields, [fieldName]: value};
+      }
+    }
+
+    return customFields;
+  }, {});
+
+export const selectCaseFieldValue = (
+  state: Partial<CaseAssistAppState>,
+  fieldName: string
+) => state?.caseField?.fields[fieldName]?.value;
+
+export const selectCustomCaseFieldValues = (
+  state: Partial<CaseAssistAppState>
+) =>
+  Object.keys(state?.caseField?.fields ?? []).reduce(
```

Comment:
Same as Sami mentioned.
```suggestion
  Object.keys(state?.caseField?.fields ?? {}).reduce(
```